integrations/SnapPixel/util.js

```diff
@@ -45,12 +48,16 @@ const eventPayload = (message) => {
 
 const ecommEventPayload = (event, message) => {
   let payload = {
-    price: get(message, "properties.price"),
+    price: parseFloat(get(message, "properties.price")),
+    client_deduplication_id: get(message, "properties.client_deduplication_id"),
     currency: get(message, "properties.currency"),
+    transaction_id:
+      get(message, "properties.transactionId") ||
+      get(message, "properties.transaction_id"),
     item_category: get(message, "properties.category"),
     description: get(message, "properties.description"),
     search_string: get(message, "properties.search_string"),
-    number_items: get(message, "properties.number_items"),
+    number_items: parseInt(get(message, "properties.number_items"), 10),
     payment_info_available: get(message, "properties.payment_info_available"),
     sign_up_method: get(message, "properties.sign_up_method"),
     success: get(message, "properties.success"),
```

Comment:
A lot of these are in common with the `eventPayload` method. Do you think we can refactor it?
CC @utsabc

Reply:
@ujjwal-ab we can put the common mapping in a function to reduce the redundant code

Reply:
Done

Reply:
Done